test/endpoint_testrunner.py

```diff
@@ -0,0 +1,238 @@
+                versions_counter[version]["inf"] = True
+            elif "tr" in tag_str:
+                versions_counter[version]["tr"] = True
+            elif "inf" in tag_str:
+                versions_counter[version]["inf"] = True
+
+            try:
+                python_version_extracted_through_regex = match.group(3)
+                if python_version_extracted_through_regex:
+                    if version not in pre_populated_py_version:
+                        pre_populated_py_version[version] = set()
+                    pre_populated_py_version[version].add(python_version_extracted_through_regex)
+            except IndexError:
+                LOGGER.info(f"For Framework: {framework} we do not use regex to fetch python version")
+
+    versions = []
+    for version, inf_train in versions_counter.items():
+        if inf_train["inf"]:
+            versions.append(version)
+
+    # Sort ascending to descending, use lambda to ensure 2.2 < 2.15, for instance
+    versions.sort(key=lambda version_str: [int(point) for point in version_str.split(".")], reverse=True)
+
+    framework_versions = versions if len(versions) < 4 else versions[:3]
+    image_definitions = []
+    for fw_version in framework_versions:
+        if fw_version in pre_populated_py_version:
+            py_versions = pre_populated_py_version[fw_version]
+        else:
+            py_versions = [get_canary_default_tag_py3_version(framework, fw_version)]
+        for py_version in py_versions:
+            image_definition = []
+            if processor == "gpu" or processor is None:
+                image_definitions.append({
+                            "registry": registry,
+                            "region": region,
+                            "repository": f"{framework}-{image_type}",
+                            "framework_version": fw_version,
+                            "processor": processor,
+                            "py_version": py_version
+                        })
+            if processor == "cpu" or processor is None:
+                image_definitions.append({
+                            "registry": registry,
+                            "region": region,
+                            "repository": f"{framework}-{image_type}",
+                            "framework_version": fw_version,
+                            "processor": processor,
+                            "py_version": py_version
+                        })
+
+    return image_definitions
+
+
+def execute_endpoint_test(framework_name, image_definition, account_id, sagemaker_region, registry, region):
+    
+    repository = image_definition.get("repository")
+    framework_version = image_definition.get("framework_version")
+    processor = image_definition.get("processor")
+    python_version = image_definition.get("py_version")
+    tag = f"{framework_version}-{processor}-{python_version}"
+
+    retries = "" # " --reruns 2"
+    instance_type = "p2.xlarge"
```

Comment:
Is this always a gpu instance type? Will only gpu images be tested?

Reply:
Yes. Only GPU images.

Reply:
Why select a p2.xlarge instance type instead of p3.2xlarge or g4dn.xlarge (all of which have 1 GPU, but the P2 instance family is extremely old).